Prod jetstream ingest failing

## Issue

Starting around 10 p.m. on Sunday, April 26, Eastern Time, Production JetStream ingest began failing. Logs: 

[jetstream-prod-downloaded-logs-20260427-151658.json](https://github.com/user-attachments/files/27144688/jetstream-prod-downloaded-logs-20260427-151658.json)

Observation in dashboard:

<img width="721" height="676" alt="Image" src="https://github.com/user-attachments/assets/d97c37e7-5f6c-4080-8e47-54494f95d869" />

Upon looking into the logs, index creation was failing during `EnsureIndex` which was introduced in #322 and refined in #339 . 

Oddly enough, it looks like the production JetStream and Megastream ingest services are creating the hourly index pattern meant for stage in addition to the weekly index pattern meant for prod:

[index-on-prod-2026-04-27.txt](https://github.com/user-attachments/files/27144714/index-on-prod-2026-04-27.txt)

This caused the ingest services to exhaust the available shard capacity on the Elasticsearch production cluster, which resulted in the failing ensure index calls:

```json
{
  "textPayload": "[3619279] [ERROR] ensureIndices failed (retrying in 1m4s): failed to ensure index for likes: create index likes-2026-w18: [400] {\"error\":{\"root_cause\":[{\"type\":\"illegal_argument_exception\",\"reason\":\"Validation Failed: 1: this action would add [14] shards, but this cluster currently has [3996]/[4000] maximum normal shards open; for more information, see https://www.elastic.co/docs/deploy-manage/production-guidance/optimize-performance/size-shards?version=9.0#troubleshooting-max-shards-open;\"}],\"type\":\"illegal_argument_exception\",\"reason\":\"Validation Failed: 1: this action would add [14] shards, but this cluster currently has [3996]/[4000] maximum normal shards open; for more information, see https://www.elastic.co/docs/deploy-manage/production-guidance/optimize-performance/size-shards?version=9.0#troubleshooting-max-shards-open;\"},\"status\":400}",
  "insertId": "69efebbe0009ffc3d7cdfc16",
  "resource": {
    "type": "cloud_run_revision",
    "labels": {
      "location": "us-east1",
      "configuration_name": "jetstream-ingest-prod",
      "service_name": "jetstream-ingest-prod",
      "project_id": "greenearth-471522",
      "revision_name": "jetstream-ingest-prod-00021-f6j"
    }
  },
  "timestamp": "2026-04-27T23:05:34.655299Z",
  "labels": {
    "instanceId": "0007b734d9d0859c580bb9c31a80d28e004f28f3dc1f4d8d8881fe888924f4cb1f10c254b222e84b8638613e8e1193725b2c846d48e9da99c79af247094abd727481b48e94c522b2fd8cfa1a72f6"
  },
  "logName": "projects/greenearth-471522/logs/run.googleapis.com%2Fstderr",
  "receiveTimestamp": "2026-04-27T23:05:34.658686598Z"
}
```



#345 introduces cleaner handling of failed web socket connections to prevent panics and service disruption, and also adds better logging for ensure index failures used above. 

## Investigation

The production services were deployed with an environment misconfiguration that caused them to trigger code creating both index patterns on the production cluster when they only should have been triggering the index pattern for the weekly index meant for prod. 

## Solution

See #347

--- a/ingest/scripts/deploy.sh
+++ b/ingest/scripts/deploy.sh
@@ -21,16 +21,6 @@ GE_AWS_S3_PREFIX="${GE_AWS_S3_PREFIX:-mega/}"
 GE_JETSTREAM_INSTANCES="${GE_JETSTREAM_INSTANCES:-1}"
 GE_MEGASTREAM_INSTANCES="${GE_MEGASTREAM_INSTANCES:-1}"
 
-# Index period: controls time-based index partitioning for posts/likes/tombstones
-# prod → week (ISO week), stage → hour, local/default → 10min
-if [ "$GE_ENVIRONMENT" = "prod" ]; then
-    GE_INDEX_PERIOD="${GE_INDEX_PERIOD:-week}"
-elif [ "$GE_ENVIRONMENT" = "stage" ]; then
-    GE_INDEX_PERIOD="${GE_INDEX_PERIOD:-hour}"
-else
-    GE_INDEX_PERIOD="${GE_INDEX_PERIOD:-10min}"
-fi
-
 # Get current git SHA (short version) for deployment tracking
 GIT_SHA=$(git rev-parse --short HEAD 2>/dev/null || echo "unknown")
 
@@ -484,9 +474,22 @@ show_service_status() {
 main() {
     local service="${1:-all}"
 
+    # Derive GE_INDEX_PERIOD from GE_ENVIRONMENT after all flags have been parsed.
+    # Using unconditional assignment so a stale GE_INDEX_PERIOD in the caller's
+    # shell cannot override the environment-appropriate value.
+    # prod → week (ISO week), stage → hour, local/default → 10min
+    if [ "$GE_ENVIRONMENT" = "prod" ]; then
+        GE_INDEX_PERIOD="week"
+    elif [ "$GE_ENVIRONMENT" = "stage" ]; then
+        GE_INDEX_PERIOD="hour"
+    else
+        GE_INDEX_PERIOD="10min"
+    fi
+
     echo "=================================================="
     echo "Green Earth Ingex - Cloud Run Source Deployment"
     echo "Environment: $GE_ENVIRONMENT"
+    echo "Index period: $GE_INDEX_PERIOD"
     echo "Project: $GE_GCP_PROJECT_ID"
     echo "Region: $GE_GCP_REGION"
     echo "Git SHA: $GIT_SHA"

